File: python/ex060.py
# Desenvolva um algoritmo que leia o nome, a idade e o sexo de várias pessoas.
# O programa vai perguntar se o usuário quer ou não continuar. No final, mostre:
# a) O nome da pessoa mais velha
# b) O nome da mulher mais jovem
# c) A média de idade do grupo
# d) Quantos homens tem mais de 30 anos
# e) Quantas mulheres tem menos de 18 anos


sumAge = 0
sumPerson = 0
inputName = ""
majorAge = 0
youngestAge  = 999
inputAge = 0
inputGender = 0
inputAnswer = 0
oldestPerson = ""
youngestWoman = ""
aboveThirty = 0
belowEighteeen = 0
avgPerson = 0



while True:
    inputName = input("Digite o seu nome: ")
    inputGender =  int(input("Digite o seu genero: (1-Homem 2-Mulher)"))
    inputAge =  int(input("Digite a sua idade: "))
    
    sumAge+= inputAge
    sumPerson+=1
    
    if(inputAge > majorAge):
        majorAge = inputAge
        
    if(inputGender == 2):
        if(inputAge < youngestAge):
            youngestAge = inputAge
        if(inputAge == youngestAge):
            youngestWoman = inputName
    
    if(inputAge == majorAge):
        oldestPerson = inputName
    
    
    if(inputGender == 1):
        if(inputAge > 30):
            aboveThirty+=1
    else:
        if(inputAge < 18):
            belowEighteeen+=1

    print("Deseja continuar inserindo? 1-Não 2-Sim")
    inputAnswer = int(input("Digite uma reposta: "))
    if(inputAnswer == 1):
        avgPerson = sumAge/sumPerson
        break
    else:
        continue
print(f" a) O nome da pessoa mais velha: {oldestPerson} \n b) O nome da mulher mais jovem: {youngestWoman} \n c) A média de idade do grupo: {avgPerson} \n d) Quantos homens tem mais de 30 anos: {aboveThirty} \n e) Quantas mulheres tem menos de 18 anos: {belowEighteeen}")

File: csharp/ex060.cs
// Desenvolva um algoritmo que leia o nome, a idade e o sexo de várias pessoas.
// O programa vai perguntar se o usuário quer ou não continuar. No final, mostre:
// a) O nome da pessoa mais velha
// b) O nome da mulher mais jovem
// c) A média de idade do grupo
// d) Quantos homens tem mais de 30 anos
// e) Quantas mulheres tem menos de 18 anos

class Ex060
{
  public string? nameOldestPerson;

  public string? nameYoungestWoman;

  public int majorAge = 0;

  public int minorAge = 999;

  public int avgAgeGroup;

  public int totalPerson = 0;

  public int countAgePerson = 0;

  public int menAboveThirty;

  public int womenBelowEighteen;

  public void FirstCountdown()
  {
    while (true)
    {
      Console.WriteLine($"Digite o seu nome: ");
      string? inputName = Console.ReadLine();

      Console.WriteLine($"Digite sua idade {inputName}: ");
      int inputAge = Convert.ToInt32(Console.ReadLine());

      Console.WriteLine($"Digite seu sexo {inputName}: 1-[sim] 2-[nao]");
      int inputGender = Convert.ToInt32(Console.ReadLine());

      totalPerson += 1;
      countAgePerson += inputAge;

      if (inputAge > majorAge)
      {
        majorAge = inputAge;
        nameOldestPerson = inputName;
      }

      if (inputGender == 2)
      {
        if (inputAge < minorAge)
        {
          minorAge = inputAge;
          nameYoungestWoman = inputName;
        }

        if (inputAge < 18)
        {
          womenBelowEighteen += 1;
        }
      }
      else
      {
        if (inputAge > 30)
        {
          menAboveThirty += 1;
        }
      }


      avgAgeGroup = countAgePerson / totalPerson;

      


      Console.WriteLine("Deseja continuar? 1-[sim] 2-[nao]");
      int inputAnswer = Convert.ToInt32(Console.ReadLine());
      if (inputAnswer == 1)
      {
        continue;
      }
      else
      {
        break;
      }
    }
    Console.Write($"nome da pessoa mais velha : {nameOldestPerson}");
    Console.Write($"nome da mulher mais jovem : {nameYoungestWoman}");
    Console.Write($"A media de idade do grupo : {avgAgeGroup}");
    Console.Write($"Quantos homens tem mais de 30 anos : {menAboveThirty}");
    Console.Write($"Quantos mulheres tem menos de 18 anos : {womenBelowEighteen}");
  }
}

